app/src/test/java/com/arielwang/workoutlogger/track/TrackViewModelTest.kt

```diff
@@ -168,63 +212,175 @@ class TrackViewModelTest {
     }
 
     @Test
-    fun `When the value of TextFiled in weight section changes, update TrackView State weightNumber`() {
+    fun `When the value of TextFiled in weightSection changes, update TrackView State weightNumber`() {
         runTest {
             val viewModel = generateViewModel()
             val onTextFieldValueChange = TrackView.Action.OnTextFieldValueChangeWeightNumber("6")
             viewModel.onUiAction(onTextFieldValueChange)
 
             viewModel.uiState.test {
                 assertEquals(
-                    TrackView.State( weightNumber = "6" ),
+                    TrackView.State(
+                        weightNumber = "6",
+                        shouldShowMaxCharAlert = ShouldShowMaxCharAlert(
+                            maxChar = 3
+                        )
+                    ),
                     awaitItem()
                 )
             }
         }
     }
 
+    // Check maximum input number of characters in weightSection TextField
```

Comment:
Better use this as the function name : Check maximum input number of characters in weightSection TextField is 3